Azure-Samples/virtual-network-integration-recipes
# Update virtual network integration for Terraform recipes.

# Purpose

This PR modifies the way virtual network integration is handled for Azure Web Apps and Functions. Instead of using the Terraform `azurerm_app_service_virtual_network_swift_connection` resource, the virtual network integration is established by setting the `virtual_network_subnet_id` property on the web app or function app.

Additionally, this PR modifies how the Terraform custom action handles caching.  Previously, the workflow may incorrectly cache the full working directory, thus preventing changes from being evaluated on tasks (such as `terraform validate`) ran against the working directory.  This PR changes the cached directory to be the working directory's __.terraform__ folder and sets the key to the GitHub SHA.  This allows for proper caching of the Terraform plugins based on the commit.

## Does this introduce a breaking change?

``` markdown
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?


``` markdown
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Get the code
* Run `terraform apply` to deploy the Azure resources.
* After the initial deployment, run `terraform plan`. There should be no changes.

_NOTE:_ If running locally (from a non-vnet connected machine), configure the deployed Azure Storage account firewall to allow access from the local machine's IP address (blocked by deny rules in the Terraform configuration).

## What to Check

Verify the virtual network integration remains for the web app (virtual network integration should not be removed).

## Other Information
### CI Pipeline runs
- [Event Hub Recipe CI](https://github.com/mcollier/virtual-network-integration-recipes/actions/runs/3913412797)
- [Private Function HTTP Recipe CI](https://github.com/mcollier/virtual-network-integration-recipes/actions/runs/3913426082)
- [Web App Private HTTP Recipe CI](https://github.com/mcollier/virtual-network-integration-recipes/actions/runs/3913415029)
- [API Management and App Gateway Recipe CI](https://github.com/mcollier/virtual-network-integration-recipes/actions/runs/3913975885)
